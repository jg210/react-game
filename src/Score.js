import React from 'react';

export const Score = props => {
  return <div>{props.score}</div>;
}

