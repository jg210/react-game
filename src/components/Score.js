// @flow
//
// (c) 2018-2019 Jeremy Green

import React from 'react';
import { connect } from "react-redux";
import type { ScoreState } from '../redux/reducers/score';

type Props = {
  score: number
}

export const Score = (props: Props) => {
  return (
    <div className="Scores">
      <div>{props.score}</div>
    </div>
  );
}

const mapStateToProps = (state: {score: ScoreState}) => {
  return { score: state.score.current };
}

export default connect(mapStateToProps)(Score);