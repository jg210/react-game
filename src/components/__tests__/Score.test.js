import React from 'react';
import ReactDOM from 'react-dom';
import { Score } from '../Score';

import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Score score="123"/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer
    .create(<Score score="123"/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});