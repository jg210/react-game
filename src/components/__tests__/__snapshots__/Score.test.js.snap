// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Scores"
  onDoubleClick={[MockFunction]}
>
  123
</div>
`;
